src/esp/assets/ResourceManager.h

```diff
@@ -966,6 +1007,14 @@ class ResourceManager {
    */
   int nextMaterialID_ = 0;
 
+  /**
+   * @brief Storage for precomuted voxel grids. Useful for when multiple objects
+   * in a scene are using the same VoxelGrid.
+   *
+   * Maps absolute path keys to VoxelGrid.
+   */
+  std::map<std::string, std::shared_ptr<esp::geo::VoxelGrid>> voxelGridDict_;
```

Comment:
What is the reasoning for having ResourceManager keep a map of voxel grids?

Is a voxel grid a "resource"? What is a "resource"? What is the job of ResourceManager? @aclegg3

Reply:
Right now I'm registering each VoxelGrid in ResourceManager so that multiple objects with the same RenderAssetHandle can share voxel grids. Do you think it'd make more sense to register the Voxel Grids in a dict in PhysicsManager?

Reply:
For this purpose the primary motivation is to keep asset related resources together. Since voxel fields are registered per-asset keyed by that asset's handle, shared per-instance, and require access to the mesh geometry this was a most straightforward place for the underlying data to live. 

At a high-level I agree this is debatable and ResourceManager owns too much overall. Let's chat more about this.